### PR TITLE
bdb: Add timeout options for bbolt

### PR DESCRIPTION
**Pull request by yyforyongyu**
A new options field, `Timeout`, which is the amount of time the bbolt waits to obtain a file lock, is added.
The option is [specified here](https://pkg.go.dev/github.com/coreos/bbolt?tab=doc#Options).

### CODE REVIEW

**Comment by Roasbeef** [comment]
Needs a rebase!

**Comment by yyforyongyu** [comment]
> Needs a rebase!

Done!

**Review by Roasbeef — APPROVED** [review]
LGTM 🛺

Post-merge, I think we'll also need to update the internal go dep mods to point to this new version of `walletdb`.

**Comment by Roasbeef** [comment]
cc @guggero

File: walletdb/example_test.go [hunk 1]
```diff
@@ -28,7 +29,7 @@ func ExampleCreate() {
 	// this, but it's done here in the example to ensure the example cleans
 	// up after itself.
 	dbPath := filepath.Join(os.TempDir(), "examplecreate.db")
-	db, err := walletdb.Create("bdb", dbPath, true)
+	db, err := walletdb.Create("bdb", dbPath, true, 10*time.Second)
```
Comment (guggero) [inline walletdb/example_test.go:32]: Should use the `defaultDBTimeout` variable here as well.

File: wallet/example_test.go [hunk 2]
```diff
@@ -32,7 +32,9 @@ func testWallet(t *testing.T) (*Wallet, func()) {
 	pubPass := []byte("hello")
 	privPass := []byte("world")
 
-	loader := NewLoader(&chaincfg.TestNet3Params, dir, true, 250)
+	loader := NewLoader(
+		&chaincfg.TestNet3Params, dir, true, 10*time.Second, 250,
```
Comment (guggero) [inline wallet/example_test.go:36]: Create and use a `defaultDBTimeout` variable for this package as well?

File: config.go [hunk 3]
```diff
@@ -33,7 +33,8 @@ const (
 	defaultRPCMaxClients    = 10
 	defaultRPCMaxWebsockets = 25
 
-	walletDbName = "wallet.db"
+	walletDbName     = "wallet.db"
```
Comment (guggero) [inline config.go:36]: While we're at it, can't we export these two variables and then use them in the `cmd/dropwtxmgr` package?

**Review by guggero — COMMENTED** [review]
LGTM, mostly nits and code consistency stuff.

File: config.go (hunk 3 above)
Comment (yyforyongyu) [reply config.go:36]: Sure. I think the params can be unified in `loader.go`, while there are multiple places defining `walletDbName`, one [here](https://github.com/btcsuite/btcwallet/blob/4c5bc1b15d3c0dc0d070ca019f4653ff95657452/wallet/loader.go#L21), one [here](https://github.com/btcsuite/btcwallet/blob/a6f3888450e4a5927dc5fd0b5c359f8fef17af78/cmd/dropwtxmgr/main.go#L31) and [here](https://github.com/btcsuite/btcwallet/blob/74a7124666b49cd7e67da87c61169f4fb83fbd73/config.go#L36).

File: wallet/watchingonly_test.go [hunk 4]
```diff
@@ -26,7 +26,9 @@ func TestCreateWatchingOnly(t *testing.T) {
 
 	pubPass := []byte("hello")
 
-	loader := NewLoader(&chaincfg.TestNet3Params, dir, true, 250)
+	loader := NewLoader(
+		&chaincfg.TestNet3Params, dir, true, 10*time.Second, 250,
```
Comment (guggero) [inline wallet/watchingonly_test.go:30]: nit: still a hard coded timeout :sweat_smile:

File: config.go (hunk 3 above)
Comment (guggero) [reply config.go]: Nice!

**Review by guggero — APPROVED** [review]
LGTM :100:

File: wallet/watchingonly_test.go (hunk 4 above)
Comment (yyforyongyu) [reply wallet/watchingonly_test.go]: Hah, my bad, now fixed 😂

**Pull request merged**